src/server/schema.ts

```diff
@@ -30,11 +30,30 @@ export const schema = `
 
   type Terms {
     source: Source!
-    terms: [Term]!
+    terms: [Term]! @deprecated(reason: "Use 'result' instead")
+    result: Result!
   }
 
   type Query {
     terms(sources: [ID]!, query: String!): [Terms]
     sources: [Source]
   }
+  
+  union Result = Success | TimeoutError | ServerError
```

Comment:
It’s not possible to say `union Result = [Term] | TimeoutError | ServerError` here.

@sdevalk Do you then still want to change the name from `Success` to `Terms`? That would give:

```graphql
... on Terms {
  terms {
    uri
    prefLabel
  }
}
```

Reply:
Good one. `... on Terms` + `terms` sounds a bit redundant.

`... on Success` and `... on Error` are nice counterparts. Would that be an option, after all? Or should we use a unique name here, e.g. `... on TermsSuccess`, to prevent conflicts with a (future) `... on Success` in the `source` section (`... on SourceSuccess`)?

Reply:
> `... on Terms` + `terms` sounds a bit redundant.

Agreed. Still, it reads okay: on receiving some terms (`... on Terms`) rather than an error (`... on Error`), unpack those terms and retrieve properties from them. 

So I decided to keep rename the type `Result` (too generic as we will have _source_ results in the future, too) to `Terms`.